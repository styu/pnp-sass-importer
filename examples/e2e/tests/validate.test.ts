import { describe, expect, it } from "vitest";
import { createRequire } from "node:module";
import fs from "node:fs";
const require = createRequire(import.meta.url);

describe("validate output", () => {
  describe("sass-example", () => {
    it("should interpret pkg: protocol correctly", () => {
      const buttonCss = require.resolve("@example/sass/css/button.css");
      const buttonCssContent = fs.readFileSync(buttonCss, "utf-8");
      expect(buttonCssContent).toMatchSnapshot();
    });

    it("should interpret relative paths and node_modules imports correctly", () => {
      const headerCss = require.resolve("@example/sass/css/header.css");
      const headerCssContent = fs.readFileSync(headerCss, "utf-8");
      expect(headerCssContent).toMatchSnapshot();
    });
  });

  describe("postcss-example", () => {
    it("should interpret pkg: protocol correctly", () => {
      const headerCss = require.resolve(
        "@example/postcss/css/button.module.css",
      );
      const headerCssContent = fs.readFileSync(headerCss, "utf-8");
      expect(headerCssContent).toMatchSnapshot();
    });

    it("should interpret relative paths and node_modules imports correctly", () => {
      const headerCss = require.resolve(
        "@example/postcss/css/header.module.css",
      );
      const headerCssContent = fs.readFileSync(headerCss, "utf-8");
      expect(headerCssContent).toMatchSnapshot();
    });
  });

  describe("postcss-cjs-example", () => {
    it("should interpret pkg: protocol correctly", () => {
      const headerCss = require.resolve(
        "@example/postcss-cjs/css/button.module.css",
      );
      const headerCssContent = fs.readFileSync(headerCss, "utf-8");
      expect(headerCssContent).toMatchSnapshot();
    });

    it("should interpret relative paths and node_modules imports correctly", () => {
      const headerCss = require.resolve(
        "@example/postcss-cjs/css/header.module.css",
      );
      const headerCssContent = fs.readFileSync(headerCss, "utf-8");
      expect(headerCssContent).toMatchSnapshot();
    });
  });
});
