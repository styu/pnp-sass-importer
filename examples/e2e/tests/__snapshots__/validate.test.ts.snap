// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`validate output > postcss-cjs-example > should interpret pkg: protocol correctly 1`] = `
"._button_1pjng_1 {
  color: #ff0000;
}"
`;

exports[`validate output > postcss-cjs-example > should interpret relative paths and node_modules imports correctly 1`] = `
"/*
 * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
 */
._header_6tek9_4 {
  color: #ff0000;
  display: flex;
  font-family: -apple-system, "BlinkMacSystemFont", "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Open Sans", "Helvetica Neue", "blueprint-icons-16", sans-serif;
  justify-content: space-between;
  max-width: 800px;
}"
`;

exports[`validate output > postcss-example > should interpret pkg: protocol correctly 1`] = `
"._button_1pjng_1 {
  color: #ff0000;
}"
`;

exports[`validate output > postcss-example > should interpret relative paths and node_modules imports correctly 1`] = `
"/*
 * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
 */
._header_6tek9_4 {
  color: #ff0000;
  display: flex;
  font-family: -apple-system, "BlinkMacSystemFont", "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Open Sans", "Helvetica Neue", "blueprint-icons-16", sans-serif;
  justify-content: space-between;
  max-width: 800px;
}"
`;

exports[`validate output > sass-example > should interpret pkg: protocol correctly 1`] = `
".button {
  color: #ff0000;
}"
`;

exports[`validate output > sass-example > should interpret relative paths and node_modules imports correctly 1`] = `
"/*
 * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
 */
.header {
  color: #ff0000;
  display: flex;
  font-family: -apple-system, "BlinkMacSystemFont", "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Open Sans", "Helvetica Neue", "blueprint-icons-16", sans-serif;
  justify-content: space-between;
  max-width: 800px;
}"
`;
